cdap-ui/app/cdap/components/AbstractWidget/FormInputs/Password/index.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { useState } from 'react';
+import TextField from '@material-ui/core/TextField';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import ThemeWrapper from 'components/ThemeWrapper';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import { IWidgetProps } from 'components/AbstractWidget';
+import { objectQuery } from 'services/helpers';
+import { WIDGET_PROPTYPES } from 'components/AbstractWidget/constants';
+
+const styles = (theme) => {
+  return {
+    input: {
+      padding: 10,
+    },
+    button: {
+      padding: 0,
+      '&:focus': {
+        outline: 'none',
+      },
+    },
+  };
+};
+
+interface IPasswordWidgetProps {
+  placeholder?: string;
+}
+
+interface IPasswordProps extends IWidgetProps<IPasswordWidgetProps>, WithStyles<typeof styles> {}
+
+function Password({ value, onChange, widgetProps, disabled, classes }: IPasswordProps) {
+  const [showPassword, setPwdVisibility] = useState<boolean>(false);
```

Comment:
nit: won't it be better to have these similar: `[showPassword ,setShowPassword]`  or `[pwdVisibility, setPwdVisibility]` or something else. So that they're easier to understand?